File: app/src/main/java/com/veganbeauty/app/features/home/adapter/HomeBannerAdapter.java
package com.veganbeauty.app.features.home.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.databinding.HomeItemBannerBinding;

import java.util.ArrayList;
import java.util.List;

public class HomeBannerAdapter extends RecyclerView.Adapter<HomeBannerAdapter.BannerViewHolder> {

    private final List<HomeBannerItem> items = new ArrayList<>();

    public void submitBanners(List<HomeBannerItem> banners) {
        items.clear();
        items.addAll(banners);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public BannerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        HomeItemBannerBinding binding = HomeItemBannerBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        );
        return new BannerViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull BannerViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class BannerViewHolder extends RecyclerView.ViewHolder {
        private final HomeItemBannerBinding binding;

        public BannerViewHolder(HomeItemBannerBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(HomeBannerItem item) {
            boolean hasOverlay = (item.getTitle() != null && !item.getTitle().trim().isEmpty()) ||
                                 (item.getActionText() != null && !item.getActionText().trim().isEmpty());
            binding.llBannerOverlay.setVisibility(hasOverlay ? View.VISIBLE : View.GONE);
            binding.tvBannerTitle.setText(item.getTitle() != null ? item.getTitle() : "");
            binding.btnBannerAction.setText(item.getActionText() != null ? item.getActionText() : "");

            if (item.getImageRes() != null) {
                binding.ivBanner.setImageResource(item.getImageRes());
            } else if (item.getImageUrl() != null && !item.getImageUrl().trim().isEmpty()) {
                com.bumptech.glide.Glide.with(binding.ivBanner.getContext()).load(item.getImageUrl()).into(binding.ivBanner);
            }
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/home/adapter/HomeBannerAdapter.kt
package com.veganbeauty.app.features.home.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import coil.load
import com.veganbeauty.app.databinding.HomeItemBannerBinding

data class HomeBannerItem(
    val imageRes: Int? = null,
    val imageUrl: String? = null,
    val title: String = "",
    val actionText: String = ""
)

class HomeBannerAdapter : RecyclerView.Adapter<HomeBannerAdapter.BannerViewHolder>() {

    private val items = mutableListOf<HomeBannerItem>()

    fun submitBanners(banners: List<HomeBannerItem>) {
        items.clear()
        items.addAll(banners)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BannerViewHolder {
        val binding = HomeItemBannerBinding.inflate(
            LayoutInflater.from(parent.context), parent, false
        )
        return BannerViewHolder(binding)
    }

    override fun onBindViewHolder(holder: BannerViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int = items.size

    class BannerViewHolder(
        private val binding: HomeItemBannerBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: HomeBannerItem) {
            val hasOverlay = item.title.isNotBlank() || item.actionText.isNotBlank()
            binding.llBannerOverlay.visibility =
                if (hasOverlay) android.view.View.VISIBLE else android.view.View.GONE
            binding.tvBannerTitle.text = item.title
            binding.btnBannerAction.text = item.actionText
            when {
                item.imageRes != null -> binding.ivBanner.setImageResource(item.imageRes)
                !item.imageUrl.isNullOrBlank() -> binding.ivBanner.load(item.imageUrl) {
                    crossfade(true)
                }
            }
        }
    }
}
